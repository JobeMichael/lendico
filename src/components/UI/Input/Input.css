.field {
  display: flex;
  align-items: center;
  position: relative;
}
.field:first-child {
  margin-right: 50px;
}
.Input {
  width: 100%;
  padding: 10px;
  box-sizing: border-box;
}
.field p {
  position: absolute;
  margin: 0;
  font-size: 12px;
  color: red;
}

.label {
  font-size: 18px;
  margin-right: 8px;
}

.input-element {
  outline: none;
  border: 2px solid #948a8a;
  background-color: white;
  font-size: 18px;
  padding: 6px 10px;
  display: block;
  width: 150px;
  box-sizing: border-box;
}

.input-element:focus {
  outline: none;
  background-color: #f3f3f3;
}

.invalid {
  border: 1px solid red;
  background-color: #fda49a;
}

/* Chrome, Safari, Edge, Opera */
input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
input[type="number"] {
  -moz-appearance: textfield;
}
.field p {
  color: #bf1650;
}

.field p::before {
  display: inline;
  content: "âš  ";
}
