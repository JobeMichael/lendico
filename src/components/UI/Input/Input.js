import React from "react";
import "./Input.css";

const input = ({ label, name, errors, inputRef, ...rest }) => (
  <div className="field">
    <label className="label">{label} : </label>
    <div className="control">
      <input
        name={name}
        autoComplete="off"
        ref={inputRef}
        className="input-element"
        {...rest}
      />
      {errors[name] && <p>{errors[name].message}</p>}
    </div>
  </div>
);

export default input;
