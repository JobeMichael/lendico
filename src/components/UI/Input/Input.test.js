import { render } from "@testing-library/react";
import React from "react";
import Input from "./Input";

const testProps = {
  label: "Amount",
  name: "amount",
  errors: {},
};

const testFormErrorProps = {
  amount: { type: "required", message: "This field required." },
};

describe("rendering", () => {
  test("renders Input correctly", () => {
    const { asFragment } = render(<Input {...testProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders Input with error", () => {
    const { asFragment } = render(
      <Input {...{ ...testProps, errors: testFormErrorProps }} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
