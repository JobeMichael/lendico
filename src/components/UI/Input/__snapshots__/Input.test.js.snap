// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders Input correctly 1`] = `
<DocumentFragment>
  <div
    class="field"
  >
    <label
      class="label"
    >
      Amount
    </label>
    <div
      class="control"
    >
      <input
        autocomplete="off"
        class="input-element"
        name="amount"
      />
      <p />
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering renders Input with error 1`] = `
<DocumentFragment>
  <div
    class="field"
  >
    <label
      class="label"
    >
      Amount
    </label>
    <div
      class="control"
    >
      <input
        autocomplete="off"
        class="input-element"
        name="amount"
      />
      <p />
    </div>
  </div>
</DocumentFragment>
`;
